pkg/theia/commands/policy_recommendation_delete.go

```diff
@@ -0,0 +1,190 @@
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		recoID, err := cmd.Flags().GetString("id")
+		if err != nil {
+			return err
+		}
+		if recoID == "" && len(args) == 1 {
+			recoID = args[0]
+		}
+		_, err = uuid.Parse(recoID)
+		if err != nil {
+			return fmt.Errorf("failed to decode input id %s into a UUID, err: %v", recoID, err)
+		}
+		kubeconfig, err := ResolveKubeConfig(cmd)
+		if err != nil {
+			return err
+		}
+		endpoint, err := cmd.Flags().GetString("clickhouse-endpoint")
+		if err != nil {
+			return err
+		}
+		if endpoint != "" {
+			_, err := url.ParseRequestURI(endpoint)
+			if err != nil {
+				return fmt.Errorf("failed to decode input endpoint %s into a url, err: %v", endpoint, err)
+			}
+		}
+		useClusterIP, err := cmd.Flags().GetBool("use-cluster-ip")
+		if err != nil {
+			return err
+		}
+
+		clientset, err := CreateK8sClient(kubeconfig)
+		if err != nil {
+			return fmt.Errorf("couldn't create k8s client using given kubeconfig, %v", err)
+		}
+
+		idMap, err := getPolicyRecommendationIdMap(clientset, kubeconfig, endpoint, useClusterIP)
+		if err != nil {
+			return fmt.Errorf("err when get policy recommendation ID map, %v", err)
+		}
+
+		if _, ok := idMap[recoID]; !ok {
+			return fmt.Errorf("could not find the policy recommendation job with given ID")
+		}
+
+		clientset.CoreV1().RESTClient().Delete().
+			AbsPath("/apis/sparkoperator.k8s.io/v1beta2").
+			Namespace(flowVisibilityNS).
+			Resource("sparkapplications").
+			Name("pr-" + recoID).
+			Do(context.TODO())
+
+		err = deletePolicyRecommendationResult(clientset, kubeconfig, endpoint, useClusterIP, recoID)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("Successfully deleted policy recommendation job with ID %s\n", recoID)
+		return nil
+	},
+}
+
+func getPolicyRecommendationIdMap(clientset kubernetes.Interface, kubeconfig string, endpoint string, useClusterIP bool) (idMap map[string]bool, err error) {
+	idMap = make(map[string]bool)
+	sparkApplicationList := &sparkv1.SparkApplicationList{}
+	err = clientset.CoreV1().RESTClient().Get().
+		AbsPath("/apis/sparkoperator.k8s.io/v1beta2").
+		Namespace(flowVisibilityNS).
+		Resource("sparkapplications").
+		Do(context.TODO()).Into(sparkApplicationList)
+	if err != nil {
+		return idMap, err
+	}
+	for _, sparkApplication := range sparkApplicationList.Items {
+		id := sparkApplication.ObjectMeta.Name[3:]
+		idMap[id] = true
+	}
+	completedPolicyRecommendationList, err := getCompletedPolicyRecommendationList(clientset, kubeconfig, endpoint, useClusterIP)
+	if err != nil {
+		return idMap, err
+	}
+	for _, completedPolicyRecommendation := range completedPolicyRecommendationList {
+		idMap[completedPolicyRecommendation.id] = true
+	}
+	return idMap, nil
+}
+
+func deletePolicyRecommendationResult(clientset kubernetes.Interface, kubeconfig string, endpoint string, useClusterIP bool, recoID string) (err error) {
+	if endpoint == "" {
```

Comment:
there seem to be a lot of code duplication with `getPolicyRecommendationResult`, could the common code be reused?

Reply:
Sure, I refactored the code setting up the ClickHouse connection among `list`, `retrieve`, and `delete` commands.